tests/integration/backward_compatible/proxy/map_nearcache_test.py

```diff
@@ -108,3 +111,71 @@ def fill_map_and_near_cache(self, count=10):
         for k, v in six.iteritems(fill_content):
             self.map.get(k)
         return fill_content
+
+
+ENTRY_COUNT = 100
+
+
+class NonStopNearCacheTest(HazelcastTestCase):
+    def setUp(self):
+        rc = self.create_rc()
+        cluster = self.create_cluster(rc, self.read_cluster_config())
+        cluster.start_member()
+
+        def event_collector():
+            events = []
+
+            def collector(e):
+                if e == LifecycleState.DISCONNECTED:
+                    events.append(e)
+
+            collector.events = events
+            return collector
+
+        collector = event_collector()
+
+        client = HazelcastClient(
+            cluster_name=cluster.id,
+            reconnect_mode=ReconnectMode.ASYNC,
+            near_caches={"map": {}},
+            lifecycle_listeners=[collector],
+        )
+
+        map = client.get_map("map").blocking()
+
+        for i in range(ENTRY_COUNT):
+            map.put(i, i)
+
+        # Populate the near cache
+        for i in range(ENTRY_COUNT):
+            map.get(i)
+
+        rc.terminateCluster(cluster.id)
+        rc.exit()
+
+        self.assertTrueEventually(lambda: self.assertEqual(1, len(collector.events)))
+
+        self.client = client
+        self.map = map
+
+    def tearDown(self):
+        self.client.shutdown()
+
+    def test_get_existing_key_from_cache_when_the_cluster_is_down(self):
+        for i in range(ENTRY_COUNT):
+            self.assertEqual(i, self.map.get(i))
+
+    def test_get_non_existing_key_from_cache_when_the_cluster_is_down(self):
+        with self.assertRaises(ClientOfflineError):
+            self.map.get(ENTRY_COUNT)
+
+    def test_put_to_map_when_the_cluster_is_down(self):
+        with self.assertRaises(ClientOfflineError):
+            self.map.put(ENTRY_COUNT, ENTRY_COUNT)
+
+    @staticmethod
+    def read_cluster_config():
+        path = os.path.abspath(__file__)
+        dir_path = os.path.dirname(path)
+        with open(os.path.join(dir_path, "hazelcast.xml")) as f:
```

Comment:
Asking just because I wonder. What is this hazelcast.xml? What kind of config it stores? And, why we don't need to create a cluster in the test in java side?(Only a single member created without a special config as far as I see )

Reply:
This is the server-side config we generally use in tests. Check the file in this folder for its content. It makes the test more resilient to batch invalidations. Note that, Python side of the Near Cache is not as complete as the Java side so a special config is needed here.